### PR TITLE
update regex for yum available updates output

### LINKED ISSUE #70949: yum - parsing issue of yum output with subsequent empty lines
##### SUMMARY
`yum` module will not parse the output of YUM properly when the output contains more than 1 subsequent empty lines and only 1 package to update.

The used regex `\n[^\w]\W+(.*)` in `re.sub` in function `parse_check_update` will merge lines separated with more than 1 subsequent empty lines.

As a result, `yum` will not recognize available update and end with no change.

If more than one update is available, the first one is not listed in module's output, but still updated.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
yum.py

##### ANSIBLE VERSION

```paste below
ansible 2.9.10
  config file = [retracted]
  configured module search path = [retracted]
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jun 11 2019, 14:33:56) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

##### OS / ENVIRONMENT

RHEL 7

##### STEPS TO REPRODUCE
**Case 1**
```yaml
      - yum:
          name: gitlab-ee
          state: latest
          enablerepo: repo_gitlab-ee
```

YUM output with one available update:
```
Loaded plugins: langpacks, product-id, search-disabled-repos, subscription-manager

This system is not registered with an entitlement server. You can use subscription-manager to register.


gitlab-ee.x86_64                                                          12.10.14-ee.0.el7                                                           repo_gitlab-ee
```
The output will get parsed into (content of variable `available_updates`):
```
['Loaded plugins: langpacks, product-id, search-disabled-repos, subscription- manager', '', 'This system is not registered with an entitlement server. You can use subscription-manager to register. gitlab-ee.x86_64                12.10.14-ee.0.el7                 repo_gitlab-ee', '']
```
This results in no recognized update. No changes done on the target system.

**Case 2**
```yaml
      - yum:
          name: gitlab-ee,td-agent
          state: latest
          enablerepo: repo_gitlab-ee
```
YUM output with multiple available updates:
```
Loaded plugins: langpacks, product-id, search-disabled-repos, subscription-manager

This system is not registered with an entitlement server. You can use subscription-manager to register.


gitlab-ee.x86_64                                                          12.10.14-ee.0.el7                                                           repo_gitlab-ee
td-agent.x86_64                                                           3.8.0-0.el7                                                                 repo_td-agent
```
The output will get parsed into (content of variable `available_updates`):
```
['Loaded plugins: langpacks, product-id, search-disabled-repos, subscription- manager', '', 'This system is not registered with an entitlement server. You can use subscription-manager to register. gitlab-ee.x86_64                12.10.14-ee.0.el7                 repo_gitlab-ee', 'td-agent.x86_64                 3.8.0-0.el7                       repo_td-agent ', '']
```
This results in gitlab-ee update not being listed in done changes, but still updated (since module is executing update with user provided list in parameter `name` anyway).

##### EXPECTED RESULTS
Proper parsed output and recognized available update by `yum` module.

##### ACTUAL RESULTS
`yum` module won't recognize available update.

### PR DESCRIPTION
**Pull request by hslee16**
##### SUMMARY

update regex in yum module to match for additional output in order to assign correct values to the `available_updates` variable.

Fixes #70949 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

yum module

##### ADDITIONAL INFORMATION

the `parse_check_update` function will now return the correctly parsed yum packages to update

### CODE REVIEW

**Comment by balonik** [comment]
After the typo there is fixed, can we get this also backported to 2.9?

**Comment by bmillemathias** [comment]
Hey @hslee16 

Thanks for the patch, I assume you tested the updated yum test broke with the not fixed yum code ?
Also could you rework the commit to add a paragraph (there is only a title If I correctly look to the commits) what it fixes and how, you can include a reference to the issue (look to https://github.com/erlang/otp/wiki/writing-good-commit-messages as an inspiration). 
Could you also add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#making-your-pr-merge-worthy) ?

Best.

**Comment by hslee16** [comment]
@bmillemathias Can do. Thanks for the informative response!

**Comment by bmillemathias** [comment]
> 
> 
> @bmillemathias Can do. Thanks for the informative response!

Thanks for the fix, should you need some help, don't hesitate to ask

**Comment by bmillemathias** [comment]
Hi @hslee16 
> @bmillemathias Can do. Thanks for the informative response!

Just wanted to make sure that everything was fine for completing the task.

Best

**Comment by hslee16** [comment]
@bmillemathias 

- added a changelog
- added an explaining commit message

File: changelogs/fragments/parse_yum_output.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+- related issue https://github.com/ansible/ansible/issues/70949
```
Comment (bmillemathias) [inline changelogs/fragments/parse_yum_output.yml]: no need a separate line, rahter put the issue URL at the end of the line hereunder between parenthesis (see other changelog files for example)

**Review by bmillemathias — CHANGES_REQUESTED** [review]

File: test/units/modules/test_yum.py [hunk 2]
```diff
@@ -128,19 +139,38 @@
 glibc.x86_64            2.17-157.el7_3.1  xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"""
 
 
-unwrapped_output_rhel7_obsoletes = unwrapped_output_rhel7 + wrapped_output_rhel7_obsoletes_postfix
```
Comment (relrod) [inline test/units/modules/test_yum.py]: Please avoid reformats like these, as they make backports harder, convolute `git blame`, and make it hard to see the actual change that you're making.

**Review by relrod — CHANGES_REQUESTED** [review]
The actual change looks fine, please avoid the test reformats though.

**Comment by relrod** [comment]
needs_info

**Comment by hslee16** [comment]
@relrod reverted formatting in tests as per your request.

File: test/units/modules/test_yum.py (hunk 2 above)
Comment (hslee16) [reply test/units/modules/test_yum.py]: reverted

**Comment by s-hertel** [comment]
Fixed in #75452

**Pull request closed without merging**